spring-cloud-stream-core-docs/src/main/asciidoc/spring-cloud-stream-overview.adoc

```diff
@@ -678,6 +655,202 @@ boolean result = pollableSource.poll(received -> {
+
+That said, in this section we’ll explain the general idea behind system level error handling and will use Rabbit binder as an example. NOTE: Kafka binder provides similar
+support although some configuration properties do differ. Also for more details and confuration options please refer to individual binder's documentation.
+
+If no internal error handlers are configured the errors are propagated to the binders and binders subsequently propagate those errors back to the messaging system.
+Depending on the capabilities of the messaging systems such system may _drop_ the message, _re-queue_ the message for re-processing or _send the failed message to DLQ_.
+Both Rabbit and Kafka support these concepts. However other binders may not, so please refer to individual binder’s documentation for details on supported system
+level error handling options.
+
+====== Drop
+
+By default if no additional system level configuration is provided the messaging system will simply drop the failed message.
+While acceptable in some cases, for most cases it is not and we need some recovery mechanism to avoid message loss.
+
+====== DLQ - Dead Letter Queue
+
+DLQ allows failed messages to be send to a special destination - _Dead Letter Queue_.
+
+When configured, failed messages will be sent to this destination for subsequent re-processing or auditing/reconciliation.
+
+For example, continuing on the previous example and to set up the DLQ with Rabbit binder you need to set the following property
+
+[source,text]
+----
+spring.cloud.stream.rabbit.bindings.input.consumer.auto-bind-dlq=true
+----
+
+Keep in mind that in the above property the word `input` corresponds to the name of the input destination binding.
+The word `consumer` states that it is a consumer property and  `auto-bind-dlq` instructs the binder to configure DLQ for `input`
+destination which results in additional Rabbit queue named `input.myGroup.dlq`.
+
+Once configured, all failed messages will be routed to this queue with error message looking similar to this:
+
+[source,text]
+----
+delivery_mode:	1
+headers:
+x-death:
+count:	1
+reason:	rejected
+queue:	input.hello
+time:	1522328151
+exchange:
+routing-keys:	input.myGroup
+Payload {"name”:"Bob"}
+----
+
+As you can see from the above, your original message is preserved for further actions.
+
+However one thing you may have noticed is that there is limited information on what was the original issue with the message processing. For example, you don’t see a stack
+trace corresponding to the original error.
+To get more relevant information about the original error you must set additional property:
+
+[source,text]
+----
+spring.cloud.stream.rabbit.bindings.input.consumer.republish-to-dlq=true
+----
+
+This will force the internal error handler to intercept the error message and add additional information to it before publishing it to DLQ.
+Once configured you will see the error message contains more information relevant to the original error.
```

Comment:
Replace with "Once configured, you can see that the error message contains more information relevant to the original error, as follows:"